homework.md

```diff
@@ -20,48 +20,217 @@ CREATE TABLE stations
 );
 ```
 
-1) Write a query that fills out this table. Try your best to pick the correct station name for each ID. You may have to make some manual choices or editing based on the inconsistencies we've found. Do try to pick the correct name for each station ID based on how popular it is in the trip data.
+1. Write a query that fills out this table. Try your best to pick the correct station name for each ID. You may have to make some manual choices or editing based on the inconsistencies we've found. Do try to pick the correct name for each station ID based on how popular it is in the trip data.
 
-Hint: your query will look something like 
+### Question 1 Answer:
 
-``` sql
-INSERT INTO stations (SELECT ... FROM tripe);
+```sql
+CREATE TABLE station_popularity
+(
+    id INT,
+    name VARCHAR(100),
+    popularity integer
+);
+
+INSERT INTO station_popularity
+SELECT Max(from_station_id) as station_id, from_station_name as station_name, count(from_station_name) as popularity
+FROM trips
+GROUP BY station_name;
+
+INSERT INTO station_popularity
+SELECT Max(to_station_id) as station_id, to_station_name as station_name, count(to_station_name) as popularity
+FROM trips
+GROUP BY station_name;
+
+CREATE TABLE partition_stations
+(
+    id INT,
+    name VARCHAR(100),
+    popularity integer,
+    row_number integer
+);
+
+INSERT INTO partition_stations
+SELECT *, ROW_NUMBER() OVER (PARTITION BY id ORDER BY popularity DESC)
+FROM station_popularity;
+
+DELETE FROM partition_stations WHERE row_number > 1;
+
+INSERT INTO stations
+SELECT id, name FROM partition_stations WHERE id IS NOT NULL;
+
+DROP TABLE station_popularity, partition_stations;
```

Comment:
I liked the iterative approach you took to answering this problem 👍 Allows more room for/visibility into error checking along the way.